sdks/go/workflow.go

```diff
@@ -0,0 +1,608 @@
+package hatchet
+
+import (
+	"context"
+	"reflect"
+	"strings"
+	"time"
+
+	contracts "github.com/hatchet-dev/hatchet/internal/services/shared/proto/v1"
+	v0Client "github.com/hatchet-dev/hatchet/pkg/client"
+	"github.com/hatchet-dev/hatchet/pkg/client/create"
+	"github.com/hatchet-dev/hatchet/pkg/client/rest"
+	"github.com/hatchet-dev/hatchet/pkg/client/types"
+	"github.com/hatchet-dev/hatchet/pkg/worker"
+	"github.com/hatchet-dev/hatchet/pkg/worker/condition"
+	"github.com/hatchet-dev/hatchet/sdks/go/internal"
+)
+
+// convertInputToType converts input (typically map[string]interface{}) to the expected struct type
+func convertInputToType(input any, expectedType reflect.Type) reflect.Value {
+	if input == nil {
+		return reflect.Zero(expectedType)
+	}
+
+	inputValue := reflect.ValueOf(input)
+	if inputValue.Type().AssignableTo(expectedType) {
+		return inputValue
+	}
+
+	// Try to convert map[string]any to the expected struct type
+	if inputMap, ok := input.(map[string]any); ok && expectedType.Kind() == reflect.Struct {
+		convertedInput := reflect.New(expectedType).Elem()
+		for i := 0; i < expectedType.NumField(); i++ {
+			field := expectedType.Field(i)
+			jsonTag := field.Tag.Get("json")
+			if jsonTag == "" {
+				jsonTag = field.Name
+			} else {
+				// Handle JSON tag options like "count,omitempty" -> "count"
+				if commaIndex := strings.Index(jsonTag, ","); commaIndex != -1 {
+					jsonTag = jsonTag[:commaIndex]
+				}
+			}
+			if val, exists := inputMap[jsonTag]; exists {
+				fieldValue := convertedInput.Field(i)
+				if fieldValue.CanSet() {
+					valReflect := reflect.ValueOf(val)
+					if valReflect.Type().AssignableTo(field.Type) {
+						fieldValue.Set(valReflect)
+					} else {
+						// Try to convert common type mismatches
+						converted, ok := convertValue(val, field.Type)
+						if ok {
+							fieldValue.Set(converted)
+						}
+					}
+				}
+			}
+		}
+		return convertedInput
+	}
+
+	return reflect.ValueOf(input)
+}
+
+// convertValue attempts to convert a value to the target type for common type mismatches
+func convertValue(val any, targetType reflect.Type) (reflect.Value, bool) {
+	valReflect := reflect.ValueOf(val)
+
+	// Handle numeric conversions (e.g., float64 -> int, int -> float64)
+	if valReflect.Kind() == reflect.Float64 && targetType.Kind() == reflect.Int {
+		if f64, ok := val.(float64); ok {
+			return reflect.ValueOf(int(f64)), true
+		}
+	}
+	if valReflect.Kind() == reflect.Float64 && targetType.Kind() == reflect.Int32 {
+		if f64, ok := val.(float64); ok {
+			return reflect.ValueOf(int32(f64)), true
+		}
+	}
+	if valReflect.Kind() == reflect.Float64 && targetType.Kind() == reflect.Int64 {
+		if f64, ok := val.(float64); ok {
+			return reflect.ValueOf(int64(f64)), true
+		}
+	}
+
+	return reflect.Value{}, false
+}
+
+// Workflow represents a workflow definition that can contain multiple tasks.
+type Workflow struct {
```

Comment:
Would also be good if `Workflow` had a `Name()` method since namespaces can modify the name